lib/common.ts

```diff
@@ -548,6 +550,15 @@ export function createChannel(streamIn: StreamIn): StreamOut {
           break;
         }
 
+        case 'serve-build': {
+          let callbacks = serveCallbacks.get(request.serveID);
+          let buildResult: types.BuildResult = { warnings: request.args.warnings }
```

Comment:
Not sure if this is ok since during `build`, `warnings` is retrieved via `replaceDetailsInMessages(response!.errors, details);`